packages/safe-core-sdk-utils/src/eip-712/index.ts

```diff
@@ -0,0 +1,70 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { GenerateTypedData, SafeTransactionEIP712Args } from '@gnosis.pm/safe-core-sdk-types'
+import semverSatisfies from 'semver/functions/satisfies'
+
+const EIP712_DOMAIN_BEFORE_V130 = [
+  {
+    type: 'address',
+    name: 'verifyingContract'
+  }
+]
+
+const EIP712_DOMAIN = [
+  {
+    type: 'uint256',
+    name: 'chainId'
+  },
+  {
+    type: 'address',
+    name: 'verifyingContract'
+  }
+]
+
+// This function returns the types structure for signing off-chain messages according to EIP-712
+function getEip712MessageTypes(safeVersion: string): {
+  EIP712Domain: typeof EIP712_DOMAIN | typeof EIP712_DOMAIN_BEFORE_V130
+  SafeTx: Array<{ type: string; name: string }>
+} {
+  const eip712WithChainId = semverSatisfies(safeVersion, '>=1.3.0')
+  return {
+    EIP712Domain: eip712WithChainId ? EIP712_DOMAIN : EIP712_DOMAIN_BEFORE_V130,
+    SafeTx: [
+      { type: 'address', name: 'to' },
+      { type: 'uint256', name: 'value' },
+      { type: 'bytes', name: 'data' },
+      { type: 'uint8', name: 'operation' },
+      { type: 'uint256', name: 'safeTxGas' },
+      { type: 'uint256', name: 'baseGas' },
+      { type: 'uint256', name: 'gasPrice' },
+      { type: 'address', name: 'gasToken' },
+      { type: 'address', name: 'refundReceiver' },
+      { type: 'uint256', name: 'nonce' }
+    ]
+  }
+}
+
+export function generateTypedData({
+  safeAddress,
+  safeVersion,
+  chainId,
+  safeTransactionData
+}: SafeTransactionEIP712Args): GenerateTypedData {
+  const eip712WithChainId = semverSatisfies(safeVersion, '>=1.3.0')
```

Comment:
I'd suggest extracting this to a constant. It's used in the function above as well.

Reply:
I don't see how a constant could apply here (`safeVersion` is a variable). Maybe a function but I don't think that is worthy

Reply:
Oh! you probably mean ">=1.3.0" :D